convert2rhel/toolopts.py

```diff
@@ -56,84 +57,80 @@ def _get_argparser():
         group.add_option("-k", "--activationkey", help="Activation key used"
-                         " for the system registration by the"
-                         " subscription-manager. It requires to have the --org"
-                         " option specified.")
+                                                       " for the system registration by the"
+                                                       " subscription-manager. It requires to have the --org"
+                                                       " option specified.")
         group.add_option("-o", "--org", help="Organization with which the"
-                         " system will be registered by the"
-                         " subscription-manager. A list of available"
-                         " organizations is possible to obtain by running"
-                         " 'subscription-manager orgs'. From the listed pairs"
-                         " Name:Key, use the Key here.")
+                                             " system will be registered by the"
+                                             " subscription-manager. A list of available"
+                                             " organizations is possible to obtain by running"
+                                             " 'subscription-manager orgs'. From the listed pairs"
+                                             " Name:Key, use the Key here.")
         group.add_option("-a", "--auto-attach", help="Automatically attach"
-                         " compatible subscriptions to the system.",
+                                                     " compatible subscriptions to the system.",
                          action='store_true')
         group.add_option("--pool", help="Subscription pool ID. If not used,"
-                         " the user is asked to choose from the available"
-                         " subscriptions. A list of the available"
-                         " subscriptions is possible to obtain by running"
-                         " 'subscription-manager list --available'.")
-        group.add_option("-v", "--variant", help="The RHEL variant to which"
-                         " the system will be converted. Available variants"
-                         " are: Server, Client, Workstation, ComputeNode. If"
-                         " not used, the user is asked to choose a variant.")
+                                        " the user is asked to choose from the available"
+                                        " subscriptionsincorret rhel variant. A list of the available"
+                                        " subscriptions is possible to obtain by running"
+                                        " 'subscription-manager list --available'.")
+        group.add_option("-v", "--variant", help="The RHEL variant params is not used anymore."
+                                                 " Convert2RHEL will user Server variant as default.")
```

Comment:
Can't we simply remove this? Or do we keep this just for the logging?

If we leave it in it will still be displayed in the help page and take up space. If there's a way to remove it but still be able to detect the argument to warn the user it would be best. Should be possible with this: https://docs.python.org/2.6/library/optparse.html#optparse.OptionParser.get_option

Otherwise I think we should just remove it and simply ignore the argument and go about the conversion.
```suggestion
```

Reply:
If we remove this options, will fail when user try to use... on line 201 we check if user trying to use the params.. if we remove this options will fail there.

so, I believe it's a good approach to have this options, and warning the user as I did.. after some versions we can remove it.

Reply:
That's why I'm asking if it's possible to use `get_option`. Getting `parsed_opts.variant` will inevitably fail if we remove this. We shouldn't have variant in help if it does nothing

Reply:
The problem is if we use `get_options` we lost the params options.. if you try to pass  you receive something like that: 
```bash
Usage: 
  convert2rhel [-h]
  convert2rhel [-u username] [-p password | -f pswd_file] [--pool pool_id | -a] [--disablerepo repoid] [--enablerepo repoid] [--serverurl url] [--no-rpm-va] [--debug] [--restart] [-y]
  convert2rhel [--disable-submgr] [--disablerepo repoid] [--enablerepo repoid] [--no-rpm-va] [--debug] [--restart] [-y]
  convert2rhel [-k key] [-o organization] [--pool pool_id | -a] [--disablerepo repoid] [--enablerepo repoid] [--serverurl url] [--no-rpm-va] [--debug] [--restart] [-y]

convert2rhel: error: no such option: --variant
```

so we need to decide if we move forward to show a user a info message, or just remove it..

Reply:
Found two possible solutions. One involves `argparse`, the successor to `optparse`, as it can suppress specific options to be displayed. Meaning it will still parse but not show in `--help` or man pages.

The other option, which I think we can go with now, is by checking if variant is in `sys.argv`. This means we only add variant to the parser if it's already part of the CLI arguments. Just remember to `import sys`.

```suggestion
        if '--variant' in sys.argv:
            group.add_option("-v", "--variant", help="The RHEL variant params is not used anymore."
                                                     " Convert2RHEL will user Server variant as default.")
```